src/components/SearchInput/index.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { useHistory } from 'react-router';
+import { Input } from './styled';
+
+const SearchInput = ({ initSearchWord, onChange }) => {
+  const [searchWord, setSearchWord] = React.useState(initSearchWord);
+  const history = useHistory();
+
+  const triggerChange = () => {
+    onChange(searchWord);
+    history.push('/');
+  };
+
+  const handleKeyDown = (event) => {
+    if (event.key === 'Enter') {
+      triggerChange();
+    }
+  };
+
+  const handleChange = (event) => {
+    setSearchWord(event.target.value);
+  };
```

Comment:
Using destructuring could be simpler:

```suggestion
  const handleChange = ({ target: { value } }) => setSearchWord(value);
```